pkg/plugins/builtin/email/email.go

```diff
@@ -70,38 +80,43 @@ func validConfig(config interface{}) error {
 func exec(stepName string, config interface{}, ctx interface{}) (interface{}, interface{}, error) {
 	cfg := config.(*Config)
 
-	parameters := struct {
-		From    string `json:"from"`
-		To      string `json:"to"`
-		Subject string `json:"subject"`
-		Body    string `json:"body"`
-	}{
-		cfg.From,
-		strings.Join(cfg.To, ","),
-		mime.BEncoding.Encode("UTF-8", cfg.Subject),
-		cfg.Body,
-	}
+	message := mail.NewMessage()
 
-	buffer := new(bytes.Buffer)
+	recipients := make([]string, len(cfg.To))
+	for i, recipient := range cfg.To {
+		recipients[i] = message.FormatAddress(recipient, "")
+	}
 
-	template := template.Must(template.New("emailTemplate").Parse(emailTemplate))
-	template.Execute(buffer, &parameters)
+	message.SetAddressHeader("From", cfg.FromAddress, cfg.FromName)
+	message.SetHeader("To", recipients...)
+	message.SetHeader("Subject", cfg.Subject)
+	message.SetBody(
+		http.DetectContentType([]byte(cfg.Body)),
+		cfg.Body,
+	)
 
-	auth := smtp.PlainAuth("", cfg.SMTPUsername, cfg.SMTPPassword, cfg.SMTPHostname)
+	port, _ := strconv.ParseUint(cfg.SMTPPort, 10, 64)
+	skipTLS, _ := strconv.ParseBool(cfg.SMTPSkipTLSVerify)
 
-	port := cfg.SMTPPort
-	if port == 0 {
-		port = 25
+	d := mail.NewDialer(cfg.SMTPHostname, int(port), cfg.SMTPUsername, cfg.SMTPPassword)
+	d.TLSConfig = &tls.Config{InsecureSkipVerify: skipTLS}
```

Comment:
You override the default config which set the ServerName to the HostName

https://github.com/go-mail/mail/blob/v2.3.1/smtp.go#L149

```suggestion
	d.TLSConfig = &tls.Config{InsecureSkipVerify: skipTLS, ServerName: cfg.SMTPHostname}
```

But do we really need an InsecureSkipVerify option ... ;)

Reply:
fixed, thanks!